pkg/proxy/ipvs/proxier.go

```diff
@@ -1455,11 +1457,15 @@ func getLocalIPs(endpointsMap proxyEndpointsMap) map[types.NamespacedName]sets.S
 	for svcPort := range endpointsMap {
 		for _, ep := range endpointsMap[svcPort] {
 			if ep.isLocal {
-				nsn := svcPort.NamespacedName
-				if localIPs[nsn] == nil {
-					localIPs[nsn] = sets.NewString()
+				// If the endpoint has a bad format, proxyutil.IPPart() will log an
```

Comment:
That's is a good fix as well. Iptables and ipvs has huge code duplication, that's why I try to refactor them to consume more identical codes. xref: #52528